classlib/src/main/java/org/teavm/classlib/java/util/TEnumMap.java

```diff
@@ -16,44 +16,34 @@
 package org.teavm.classlib.java.util;
 
 import java.io.Serializable;
-import java.util.AbstractMap;
-import java.util.AbstractSet;
 import java.util.Arrays;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.NoSuchElementException;
-import java.util.Objects;
-import java.util.Set;
+import org.teavm.classlib.java.lang.TCloneNotSupportedException;
+import org.teavm.interop.Rename;
 
-public class TEnumMap<K extends Enum<K>, V> extends AbstractMap<K, V> implements Serializable, Cloneable {
+public class TEnumMap<K extends Enum<K>, V> extends TAbstractMap<K, V> implements Serializable, Cloneable {
     private Class<K> keyType;
     private Object[] data;
     private boolean[] provided;
     private int size;
-    private Set<Entry<K, V>> entrySet;
+    private TSet<Entry<K, V>> entrySet;
 
     public TEnumMap(Class<K> keyType) {
-        initFromKeyType(keyType);
+        initFromKeyType(TObjects.requireNonNull(keyType));
     }
 
     public TEnumMap(TEnumMap<K, V> m) {
         initFromOtherEnumMap(m);
     }
 
-    public TEnumMap(Map<K, V> m) {
+    public TEnumMap(TMap<K, V> m) {
         if (m instanceof TEnumMap) {
             initFromOtherEnumMap((TEnumMap<K, V>) m);
         } else {
             if (m.isEmpty()) {
                 throw new IllegalArgumentException();
             }
             initFromKeyType(m.keySet().iterator().next().getDeclaringClass());
-            for (Entry<K, V> entry : m.entrySet()) {
-                int index = entry.getKey().ordinal();
-                provided[index] = true;
-                data[index] = entry.getValue();
-            }
-            size = m.size();
+            putAll(m);
```

Comment:
Isn't old implementation more efficient?

Reply:
It fails when we put some map like Map.of(Size.small, 1, Color.green, 2)

Reply:
So may be it was reasonable to fix old implementation?